core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -8,6 +8,22 @@ import {
   SsrOptimizationOptions,
 } from './ssr-optimization-options';
 
+export type SsrCallbackFn = (
+  /**
+   * Error that might've occurred while rendering.
+   */
+  err?: Error | null | undefined,
+  /**
+   * HTML response.
+   */
+  html?: string | undefined,
+  /**
+   * The indicator if `this.callbackQueue` is being processed.
+   * Mitigates the infinite loop.
+   */
+  queueProcessing?: boolean
```

Comment:
I'd be cautious with adding a custom parameter in this callback signature. This callback signature is defined by the `express` library and potentially it may add a new optional 3rd parameter there in the future versions.

![image](https://user-images.githubusercontent.com/4001059/132501486-ea246184-e0f5-4036-afcc-4ea696d3842e.png)

Do you see other options to avoid an infinite loop?

Reply:
Unfortunately, I don't see any other way to prevent the infinite loop.
Maybe we can use double underscore to prefix the variable?
Additionally, I added the additional typesafety to server.ts